financial-templates-lib/clients/ExpiringMultiPartyClient.js

```diff
@@ -91,31 +91,33 @@ class ExpiringMultiPartyClient {
   }
 
   async update() {
-    // Since this function can have a relatively long run-time, we should be safe and check contract state explicitly at the same block height.
-    // This way we won't experience errors where the contract state changes between calls.
-    this.latestBlock = await this.web3.eth.getBlockNumber();
-
-    this.collateralRequirement = this.toBN(
-      (await this.emp.methods.collateralRequirement().call(undefined, this.latestBlock)).toString()
-    );
-    this.liquidationLiveness = Number(await this.emp.methods.liquidationLiveness().call(undefined, this.latestBlock));
-
-    const events = await this.emp.getPastEvents("NewSponsor", { fromBlock: 0, toBlock: this.latestBlock });
+    // Fetch contract state variables in parallel.
+    const [collateralRequirement, liquidationLiveness, events, cumulativeFeeMultiplier] = await Promise.all([
+      this.emp.methods.collateralRequirement().call(),
+      this.emp.methods.liquidationLiveness().call(),
+      this.emp.getPastEvents("NewSponsor", { fromBlock: 0 }),
+      this.emp.methods.cumulativeFeeMultiplier().call()
+    ]);
+    this.collateralRequirement = this.toBN(collateralRequirement.toString());
+    this.liquidationLiveness = Number(liquidationLiveness);
     this.sponsorAddresses = [...new Set(events.map(e => e.returnValues.sponsor))];
-
-    // Fetch information about each sponsor.
-    const positions = await Promise.all(
-      this.sponsorAddresses.map(address => this.emp.methods.positions(address).call(undefined, this.latestBlock))
-    );
-    const collateral = await Promise.all(
-      this.sponsorAddresses.map(address => this.emp.methods.getCollateral(address).call(undefined, this.latestBlock))
+    this.cumulativeFeeMultiplier = this.toBN(cumulativeFeeMultiplier.toString());
+
+    // Fetch sponsor position, liquidation, and current time data in parallel.
+    const res = await Promise.all(
+      this.sponsorAddresses
+        .map(address => this.emp.methods.positions(address).call())
+        .concat(this.sponsorAddresses.map(address => this.emp.methods.getLiquidations(address).call()))
+        .concat(this.emp.methods.getCurrentTime().call())
     );
+    const positions = res.slice(0, this.sponsorAddresses.length);
+    const allLiquidations = res.slice(this.sponsorAddresses.length, res.length - 1);
+    const currentTime = res[res.length - 1];
```

Comment:
nit: might be easier to read if you used nested Promise.alls:
```js
    const [positions, allLiquidations, currentTime] = await Promise.all(
      [
         Promise.all(this.sponsorAddresses.map(address => this.emp.methods.positions(address).call())),
         Promise.all(this.sponsorAddresses.map(address => this.emp.methods.getLiquidations(address).call())),
         this.emp.methods.getCurrentTime().call()
      ]
    );

Reply:
Doesn't this change the one parallel call to three different parallel calls?

Reply:
Since we're not awaiting the nested promises, they should only block after all of them are fired off and you await the surrounding one (if I understand correctly). All the Promise.all should be doing is grouping an array of promises into a single one that resolves when all of its constituents do, but it shouldn't block unless awaited.

Reply:
Cool,  done